src/io/trajectory.cpp

```diff
@@ -0,0 +1,98 @@
+#include "aikido/io/trajectory.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <boost/program_options.hpp>
+#include "aikido/common/Spline.hpp"
+#include "aikido/io/yaml.hpp"
+#include "aikido/io/detail/yaml_extension.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+#include "aikido/trajectory/Interpolated.hpp"
+
+using aikido::statespace::ConstStateSpacePtr;
+using aikido::statespace::StateSpacePtr;
+using aikido::statespace::dart::MetaSkeletonStateSpacePtr;
+using aikido::trajectory::Spline;
+using aikido::trajectory::UniqueSplinePtr;
+
+namespace aikido {
+namespace io {
+
+void saveTrajectory(const aikido::trajectory::Spline& trajectory,
+    const MetaSkeletonStateSpacePtr& skelSpace, const std::string& savePath)
+{
+  std::ofstream file(savePath);
+  YAML::Emitter emitter;
+
+  statespace::ConstStateSpacePtr trajSpace = trajectory.getStateSpace();
+  emitter << YAML::BeginMap;
+  emitter << YAML::Key << "configuration" << YAML::BeginMap;
+  emitter << YAML::Key << "start_time" << YAML::Value << trajectory.getStartTime();
+  emitter << YAML::Key << "dofs" << YAML::Flow << skelSpace->getProperties().getDofNames();
+  emitter << YAML::Key << "type" << YAML::Value << "spline";
+  emitter << YAML::Key << "spline";
+  emitter << YAML::BeginMap;
+  emitter << YAML::Key << "order" << YAML::Value << trajectory.getNumDerivatives();
+  emitter << YAML::EndMap;
+  emitter << YAML::EndMap << YAML::Key << "data" << YAML::BeginSeq;
```

Comment:
Put the `YAML::Key << "data" << YAML::BeginSeq` on two more lines? I'd also move this right before the for loop.